lib/models/device.ts

```diff
@@ -1288,17 +1289,64 @@ const getDeviceModel = function (
 		 * });
 		 *
 		 * @example
+		 * balena.models.device.getSupervisorTargetState(123, 3).then(function(state) {
+		 * 	console.log(state);
+		 * });
+		 *
+		 * @example
 		 * balena.models.device.getSupervisorTargetState('7cf02a6', function(error, state) {
 		 * 	if (error) throw error;
 		 * 	console.log(state);
 		 * });
 		 */
 		getSupervisorTargetState: async (
 			uuidOrId: string | number,
+			version?: number,
 		): Promise<DeviceState.DeviceState> => {
 			const { uuid } = await exports.get(uuidOrId, { $select: 'uuid' });
 			const { body } = await request.send({
-				url: `/device/v2/${uuid}/state`,
+				url: `/device/v${version !== 3 ? 2 : 3}/${uuid}/state`,
+				baseUrl: apiUrl,
+			});
+			return body;
+		},
+
+		/**
+		 * @summary Get the target supervisor state on a "generic" device on a fleet
+		 * @name getSupervisorTargetStateForFleet
+		 * @public
+		 * @function
+		 * @memberof balena.models.device
+		 *
+		 * @param {String|Number} uuidOrId - fleet uuid (string) or id (number)
+		 * @param {String} release - (optional) release uuid (default tracked)
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.device.getSupervisorTargetStateForFleet('7cf02a6').then(function(state) {
+		 * 	console.log(state);
+		 * });
+		 *
+		 * @example
+		 * balena.models.device.getSupervisorTargetStateForFleet(123).then(function(state) {
+		 * 	console.log(state);
+		 * });
+		 *
+		 * @example
+		 * balena.models.device.getSupervisorTargetStateForFleet(123, '7cf02a6').then(function(state) {
+		 * 	console.log(state);
+		 * });
+		 *
+		 */
+		getSupervisorTargetStateForApp: async (
```

Comment:
Since that's more related to the app rather than the device, how about moving it under the application model like: `balena.models.application.getSupervisorTargetState`?

Reply:
@thgreasi 
On the API it's attached to the device endpoint as it represent the "virtual target state" of a "future" device on fleet.

For coherence between the two, I'd prefer to keep it on device.